docs/custom-transactions.asciidoc

```diff
@@ -30,25 +30,35 @@ var sqs = require('simple-sqs')()
 var queue = sqs(queueUrl, function (msg, cb) {
   // The SQS queue will send multiple messages using an array
   // of records
-  msg.Body.Records.forEach(function (job) {
-    // start one new transaction for each job record received
-    // on the queue
-    var name = 'Job ' + job.type
-    var type = 'job'
-    var trans = apm.startTransaction(name, type)
-
-    // call the function that actually processes the job
-    processJob(job, function (err) {
-      // if the job could not be processes, set the result to
-      // a 5xx error code. Here 500 indicates error, 200 is ok
-      trans.result = err ? 500 : 200
-
-      // end the transaction
-      trans.end()
-
-      cb(err)
+  var tasks = msg.Body.Records.map(function (job) {
+    return new Promise(function (resolve, reject) {
+      // start one new transaction for each job record received
+      // on the queue
+      var name = 'Job ' + job.type
+      var type = 'job'
+      var trans = apm.startTransaction(name, type)
+
+      // call the function that actually processes the job
+      processJob(job, function (err) {
+        // if the job could not be processes, set the result to
+        // a 5xx error code. Here 500 indicates error, 200 is ok
+        trans.result = err ? 500 : 200
+
+        // end the transaction
+        trans.end()
+
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
     })
   })
+
+  Promise.all(tasks).then(function () {
```

Comment:
Any reason to not just use `cb` directly stead of the nested `function`?

Reply:
Just to avoid passing values into the error argument. I suppose we resolve without an argument though, so maybe it'd _technically_ be fine. It's just not good practice to pass error-first callbacks directly into promise resolve callbacks though. 🤔

Reply:
Ah ok, makes sense.